core/src/main/java/org/aerogear/mobile/core/utils/AppIdGenerator.java

```diff
@@ -0,0 +1,44 @@
+package org.aerogear.mobile.core.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.aerogear.mobile.core.MobileCore;
+
+import java.util.UUID;
+
+/**
+ * Helper for generating ID's
+ */
+public class AppIdGenerator {
```

Comment:
-1 to the name.  Generally when I see appIds they are in the context of a identifier for an application for a service and is shared for all users of the application on that platform.

Reply:
Yeah. Going to rename thay. . I will probably want to avoid static method as well

Reply:
Actually, this is not about the App Id but the Client ID. We differentiate between them in the metrics (App Id is just the package name where the Client ID is the randomly generated and stored identifier). This could also be moved to a method in core, like `core.getDeviceId()`